dev/update_requirements.py

```diff
@@ -0,0 +1,118 @@
+"""
+This script updates the `max_major_version` attribute of each package in a YAML dependencies
+specification (e.g. requirements/core-requirements.yaml) to the maximum available version on PyPI.
+"""
+
+import argparse
+import difflib
+import re
+
+import yaml
+import requests
+from packaging.version import Version, InvalidVersion
+
+
+def get_latest_major_version(package_name: str, minium_version: str = None) -> int:
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    response = requests.get(url)
+    response.raise_for_status()
+    data = response.json()
+    min_ver = Version(minium_version if minium_version else "0.0.0")
+    versions = []
+    for version, distributions in data["releases"].items():
+        if len(distributions) == 0 or any(d.get("yanked", False) for d in distributions):
+            continue
+
+        try:
+            version = Version(version)
+        except InvalidVersion:
+            # Ignore invalid versions such as https://pypi.org/project/pytz/2004d
+            continue
+
+        if version.is_devrelease or version.is_prerelease or version < min_ver:
+            continue
+
+        versions.append(version)
+
+    return max(versions).major + 1  # +1 for testing, will be removed
+
+
+def replace_max_major_version(yaml_string: str, pip_release: str, max_major_version: int) -> str:
+    pattern = r"""
+  pip_release: {pip_release}
+  (.*?)max_major_version: \d+
+""".format(
+        pip_release=pip_release
+    )
+    repl = r"""
+  pip_release: {pip_release}
+  \1max_major_version: {max_major_version}
+""".format(
+        pip_release=pip_release, max_major_version=max_major_version
+    )
+    assert re.search(pattern, yaml_string, flags=re.DOTALL)
+    return re.sub(pattern, repl, yaml_string, flags=re.DOTALL)
```

Comment:
Seems like it might be less error prone to parse the yaml using `yaml.safe_load()` to obtain a dictionary and then mutate dictionary entries, right? Worried that this regex approach might break (e.g. imagine the pip release and max major version fields are swapped for some entry).

Reply:
> to obtain a dictionary and then mutate dictionary entries, right?

We can, but this approach doesn't preserve the comments in the YAML file:

```yaml
flask:
  # Comment 1
  pip_release: Flask
  # Comment 2
  max_major_version: 2
```

We can start with the dict approach and then switch to the regex approach when necessary.